File: java/solutions/02-intermediate/ex037_generic_box/Box.java
package fewolearning.exercises.intermediate.ex037_generic_box;

/*
Exercise 037 - Generic box (reference solution).
*/
public final class Box<T> {
    private T value;
    private boolean present;

    public void set(T value) {
        this.value = value;
        this.present = true;
    }

    public T get() {
        if (!present) {
            throw new IllegalStateException("box is empty");
        }
        return value;
    }

    public boolean isEmpty() {
        return !present;
    }
}

File: kotlin/solutions/01-beginner/ex025_generic_box/Box.kt
package fewolearning.exercises.beginner.ex025_generic_box

/*
Exercise 025 - Generic box (reference solution).
*/
class Box<T>(private var value: T? = null) {
    fun set(value: T) {
        this.value = value
    }

    fun get(): T? = value
}
